econml/sklearn_extensions/ensemble.py

```diff
@@ -0,0 +1,722 @@
+    ``max_features=n_features``, if the improvement
+    of the criterion is identical for several splits enumerated during the
+    search of the best split. To obtain a deterministic behaviour during
+    fitting, ``random_state`` has to be fixed.
+
+    The default value ``max_features="auto"`` uses ``n_features``
+    rather than ``n_features / 3``. The latter was originally suggested in
+    [1]_, whereas the former was more recently justified empirically in [2]_.
+
+    References
+    ----------
+
+    .. [1] L. Breiman, "Random Forests", Machine Learning, 45(1), 5-32, 2001.
+
+    .. [2] P. Geurts, D. Ernst., and L. Wehenkel, "Extremely randomized
+           trees", Machine Learning, 63(1), 3-42, 2006.
+
+    .. [3] S. Athey, S. Wager, "Estimation and Inference of Heterogeneous Treatment Effects using Random Forests",
+            Journal of the American Statistical Association 113.523 (2018): 1228-1242.
+
+    .. [4] S. Athey, J. Tibshirani, and S. Wager, "Generalized random forests",
+            The Annals of Statistics, 47(2), 1148-1178, 2019.
+
+    """
+
+    def __init__(self,
+                 n_estimators=100,
+                 criterion="mse",
+                 max_depth=None,
+                 min_samples_split=2,
+                 min_samples_leaf=1,
+                 min_weight_fraction_leaf=0.,
+                 max_features="auto",
+                 max_leaf_nodes=None,
+                 min_impurity_decrease=0.,
+                 subsample_fr='auto',
+                 honest=True,
+                 n_jobs=None,
+                 random_state=None,
+                 verbose=0,
+                 warm_start=False):
+        super(SubsampledHonestForest, self).__init__(
+            base_estimator=DecisionTreeRegressor(),
+            n_estimators=n_estimators,
+            estimator_params=("criterion", "max_depth", "min_samples_split",
+                              "min_samples_leaf", "min_weight_fraction_leaf",
+                              "max_features", "max_leaf_nodes",
+                              "min_impurity_decrease",
+                              "random_state"),
+            bootstrap=False,
+            oob_score=False,
+            n_jobs=n_jobs,
+            random_state=random_state,
+            verbose=verbose,
+            warm_start=warm_start)
+
+        self.n_estimators = n_estimators
+        self.criterion = criterion
+        self.max_depth = max_depth
+        self.min_samples_split = min_samples_split
+        self.min_samples_leaf = min_samples_leaf
+        self.min_weight_fraction_leaf = min_weight_fraction_leaf
+        self.max_features = max_features
+        self.max_leaf_nodes = max_leaf_nodes
+        self.min_impurity_decrease = min_impurity_decrease
+        self.subsample_fr = subsample_fr
+        self.honest = honest
+        self.estimators_ = None
+        self.vars_ = None
+
+        return
+
+    def fit(self, X, y, sample_weight=None, sample_var=None):
+        """

```

Comment:
This docstring is not properly formatted:
* It's missing the description
* The parameter descriptions are put where their types belong

Reply:
Done